[SYCL][Doc] Add slm_per_subslice query for Xe GPUs

Pennycook opened the pull request:
Extends sycl_ext_intel_device_info with a new query for the amount of SLM per subslice. The amount of SLM per subslice may be greater than the amount of memory available to a single work-group, and can be used to reason about occupancy.

Pennycook commented:
@gmlueck - I've opened this as a draft because I think that's your preferred workflow for updates to supported extensions, but this is ready for review.

bader commented:
@gmlueck, Intel introduced new terminology for GPUs in 2022. 

https://www.intel.com/content/www/us/en/docs/oneapi/optimization-guide-gpu/2023-0/intel-iris-xe-gpu-architecture.html#XE-TERMINOLOGY-CHANGES

I think we should update DPC++ documentation/specifications to use new terminology at some point.

gmlueck commented:
> I think we should update DPC++ documentation/specifications to use new terminology at some point.

The APIs in this extension are all pass-throughs to Level Zero.  It seems like the Level Zero documentation is still using the old terminology:

https://oneapi-src.github.io/level-zero-spec/level-zero/latest/core/api.html#_CPPv422ze_device_properties_t

AlexeySachkov commented:
> > I think we should update DPC++ documentation/specifications to use new terminology at some point.
> 
> The APIs in this extension are all pass-throughs to Level Zero. It seems like the Level Zero documentation is still using the old terminology:
> 
> https://oneapi-src.github.io/level-zero-spec/level-zero/latest/core/api.html#_CPPv422ze_device_properties_t

But do our customers care about whether or not this is a pass-through to L0? We already have asks to update the used terminology: #16068

Pennycook committed:
[SYCL][Doc] Add slm_per_subslice query for Xe GPUs

Extends sycl_ext_intel_device_info with a new query for the amount of SLM per
subslice. The amount of SLM per subslice may be greater than the amount of
memory available to a single work-group, and can be used to reason about
occupancy.

Signed-off-by: John Pennycook <john.pennycook@intel.com>

gmlueck commented:
Reopening and removing "stale" label because implementation of this change is still on the backlog as CMPLRLLVM-64348.  Once this API is implemented, we can merge this spec PR.
